core/embed/firmware/bl_check.c

```diff
@@ -97,6 +99,50 @@ void check_and_replace_bootloader(void) {
       (const uint32_t *)&_binary_embed_firmware_bootloader_bin_start;
   const uint32_t len =
       (const uint32_t)&_binary_embed_firmware_bootloader_bin_size;
+
+  image_header new_bld_hdr;
+  secbool new_header_valid =
+      parse_image_header((uint8_t *)data, BOOTLOADER_IMAGE_MAGIC,
+                         BOOTLOADER_IMAGE_MAXSIZE, &new_bld_hdr);
+
+  if (new_header_valid != sectrue) {
+    // invalid bootloader header
+    // could also mean that the embedded bootloader is for another model
+    return;
+  }
+
+  image_header current_bld_hdr;
+  secbool old_header_valid =
+      parse_image_header(bl_data, BOOTLOADER_IMAGE_MAGIC,
+                         BOOTLOADER_IMAGE_MAXSIZE, &current_bld_hdr);
+
+  (void)old_header_valid;
```

Comment:
Can't think of a scenario when this would be significant.

Even when using pre-check bootloader on TT devboard and flashing production TR firmware via st-link/j-link, this still shouldn't run because of the expiry mismatch. 

Otoh maybe there is some scenario a can't imagine right now, so i will introduce the check here. Unless we can think of scenario where overwriting invalid bootloader header would actually fix things. Like if bootloader runs the firmware, gets damaged somehow and this would repair it.. that probably can't happen though

Reply:
my first thought was along the lines of "we are on a TT hardware, the installed bootloader is for TR, and somehow we managed to load TT firmware with the correct bootloader in it.

Reply:
but yeah, that's super unlikely to happen through all the other measures that we have here.

Reply:
added the check: https://github.com/trezor/trezor-firmware/commit/df3eccea7ad47dae23ec6f4c1d67990f19a3e39b#diff-64358366e8b14d6e519d7910beaa9a67feeb32d9b8f728456182cd3ac3be6d5d